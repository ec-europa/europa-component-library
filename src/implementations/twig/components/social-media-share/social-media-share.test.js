import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe } from 'vitest-axe';

import demoData from '@ecl/specs-component-social-media-share/demo/data';

describe('Social Media Share', () => {
  const template = '@ecl/social-media-share/social-media-share.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);
  describe('Default', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(demoData, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(demoData, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, demoData, true)),
      ).toHaveNoViolations();
    });
  });
  describe('Default view', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(demoData)).resolves.toMatchSnapshot();
    });
  });
});
