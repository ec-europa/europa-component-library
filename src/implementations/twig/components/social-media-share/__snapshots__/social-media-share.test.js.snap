// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Media Share Default renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-social-media-share"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#twitter-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#facebook-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#linkedin-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#telegram-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <div
          class="ecl-popover"
          data-ecl-auto-init="Popover"
        >
          <a
            aria-controls="social-media-share-popover"
            aria-expanded="false"
            class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__toggle ecl-social-media-share__link"
            data-ecl-popover-toggle=""
            href="/example"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#share"
              />
            </svg>
            <span
              class="ecl-link__label"
            >
              Other social networks
            </span>
          </a>
          <div
            class="ecl-popover__container"
            hidden=""
            id="social-media-share-popover"
          >
            <div
              class="ecl-popover__content"
            >
              <ul
                class="ecl-popover__list"
              >
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#pinterest-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Pinterest
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#mastodon-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Mastodon
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#reddit-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Reddit
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#youtube-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Youtube
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#flickr-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Flickr
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#skype-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Skype
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#spotify-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Spotify
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Social Media Share Default renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-social-media-share custom-class custom-class--test"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#twitter-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#facebook-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#linkedin-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#telegram-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <div
          class="ecl-popover"
          data-ecl-auto-init="Popover"
        >
          <a
            aria-controls="social-media-share-popover"
            aria-expanded="false"
            class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__toggle ecl-social-media-share__link"
            data-ecl-popover-toggle=""
            href="/example"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#share"
              />
            </svg>
            <span
              class="ecl-link__label"
            >
              Other social networks
            </span>
          </a>
          <div
            class="ecl-popover__container"
            hidden=""
            id="social-media-share-popover"
          >
            <div
              class="ecl-popover__content"
            >
              <ul
                class="ecl-popover__list"
              >
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#pinterest-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Pinterest
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#mastodon-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Mastodon
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#reddit-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Reddit
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#youtube-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Youtube
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#flickr-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Flickr
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#skype-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Skype
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#spotify-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Spotify
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`Social Media Share Default view renders correctly 1`] = `
<jest>
  <div
    class="ecl-social-media-share"
  >
    <p
      class="ecl-social-media-share__description"
    >
      Share this page
    </p>
    <ul
      class="ecl-social-media-share__list"
    >
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#twitter-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#facebook-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Facebook
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#linkedin-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Linkedin
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <a
          class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-social-media-share__link"
          href="/example"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
            focusable="false"
          >
            <use
              xlink:href="/icon-social-media.svg#telegram-color"
            />
          </svg>
          <span
            class="ecl-link__label"
          >
            Telegram
          </span>
        </a>
      </li>
      <li
        class="ecl-social-media-share__item"
      >
        <div
          class="ecl-popover"
          data-ecl-auto-init="Popover"
        >
          <a
            aria-controls="social-media-share-popover"
            aria-expanded="false"
            class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__toggle ecl-social-media-share__link"
            data-ecl-popover-toggle=""
            href="/example"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m ecl-link__icon ecl-social-media-share__icon"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#share"
              />
            </svg>
            <span
              class="ecl-link__label"
            >
              Other social networks
            </span>
          </a>
          <div
            class="ecl-popover__container"
            hidden=""
            id="social-media-share-popover"
          >
            <div
              class="ecl-popover__content"
            >
              <ul
                class="ecl-popover__list"
              >
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#pinterest-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Pinterest
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#mastodon-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Mastodon
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#reddit-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Reddit
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#youtube-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Youtube
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#flickr-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Flickr
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#skype-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Skype
                    </span>
                  </a>
                </li>
                <li
                  class="ecl-popover__item"
                >
                  <a
                    class="ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-popover__link"
                    href="/example"
                  >
                    <svg
                      aria-hidden="true"
                      class="ecl-icon ecl-icon--s ecl-link__icon"
                      focusable="false"
                    >
                      <use
                        xlink:href="/icon-social-media.svg#spotify-color"
                      />
                    </svg>
                    <span
                      class="ecl-link__label"
                    >
                      Spotify
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</jest>
`;
